beancount/query/query_compile.py

```diff
@@ -90,6 +91,16 @@ def __init__(self, value):
     def __call__(self, _):
         return self.value
 
+class EvalAmount(EvalNode):
+    __slots__ = ('decimal','currency')
+
+    def __init__(self, decimal,currency):
+        super().__init__(type(amount.Amount))
+        self.decimal = decimal
+        self.currency = currency
+
+    def __call__(self, _):
+        return amount.Amount(self.decimal,self.currency)
```

Comment:
Missing space after `,`. Also, wouldn't it be simpler to have a single `value` class member, construct the `Amount` instance in the `__init__` method and assign it to `self.value` and just return it here?

Reply:
code is removed.